Sources/Harbor/HRequestManager.swift

```diff
@@ -27,27 +30,36 @@ extension HRequestManager {
                 try? await Task.sleep(nanoseconds: delayInNanoseconds)
             }
 
-            if let error = mock.error {
-                return .error(error)
+            if let hError = mock.error {
+                logError(hError, request: request)
+                return .error(hError)
             }
 
             let data = mock.jsonResponse?.data(using: .utf8) ?? Data()
             return await HRequestManager.processResponse(model: model, request: request, statusCode: mock.statusCode, data: data)
         }
 
         if !self.isConnectedToNetwork() {
-            return .error(.noConnectionError)
+            let hError: HRequestError = .noConnectionError
+            logError(hError, request: request)
```

Comment:
Maybe we should add a config so the user can turn this logs off, what if they already have some logs on?

Reply:
All the logs are executed only if HDebugRequestProtocol is implemented.